### PR TITLE
Fixes #29537: Use puppet AIO in development and testing

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,10 +8,14 @@ addons:
     - xsltproc
     - docbook-xsl
     - ca-certificates
-rvm: 2.4
-env:
-  - PUPPET_VERSION=5.0
+before_script:
+  - wget https://apt.puppetlabs.com/puppet5-release-bionic.deb
```
Comment (ekohl) [inline .travis.yml]: Let's update to 6 while we're at it.

File: Gemfile [hunk 2]
```diff
@@ -2,12 +2,6 @@ source 'https://rubygems.org'
...
 gem 'kafo', '~> 4.1'
 gem 'librarian-puppet'
-gem 'puppet', ENV.key?('PUPPET_VERSION') ? "~> #{ENV['PUPPET_VERSION']}" : '~> 6.0'
-if RUBY_VERSION < '2.1'
-  gem 'puppet-strings', '< 2.0.0'
-else
-  gem 'puppet-strings'
-end
 gem 'rake'
 gem 'rdoc', '< 6' if RUBY_VERSION < '2.2'
```
Comment (ekohl) [inline Gemfile]: Unrelated, but we can drop this since we dropped Ruby 2. Same for `semverse` but I guess that can go in a different PR as well.

**Comment by ekohl** [comment]
I think https://ci.theforeman.org/job/foreman-installer-develop-source-release/ will continue to work since Puppet is installed system wide anyway.

**Comment by ekohl** [comment]
And that was the wrong button.

File: README.md [hunk 3]
```diff
@@ -20,6 +20,12 @@ See the manual and quickstart guide available at [theforeman.org](http://thefore
 
 git clone --recursive git://github.com/theforeman/foreman-installer.git -b develop
 
+Using the git version for develop or testing requires the puppet-agent AIO package be available. Follow the instructions for your OS from [Puppet](https://puppet.com/docs/puppet/5.5/install_linux.html). Additionally, after installing puppet-agent the puppet-strings gem needs to be available in the environment:
```
Comment (ekohl) [inline README.md]: Should we use latest instead of a version?

File: README.md [hunk 4]
```diff
@@ -20,6 +20,12 @@ See the manual and quickstart guide available at [theforeman.org](http://thefore
 
 git clone --recursive git://github.com/theforeman/foreman-installer.git -b develop
```
Comment (ekohl) [inline README.md]: This is sufficient nowadays:
```suggestion
git clone https://github.com/theforeman/foreman-installer
```

**Review by ekohl — APPROVED** [review]

**Comment by ehelms** [comment]
This got too crazy in terms of support. Closing in favor of fixing librarianp.

**Pull request closed without merging**